scripts/install.sh

```diff
@@ -69,31 +81,118 @@ printf "Network: $NETWORK\n"
 
 cd $KEEP_BEACON_SOL_PATH
 
-printf "${LOG_START}Installing YARN dependencies...${LOG_END}"
+printf "${LOG_START}Installing beacon YARN dependencies...${LOG_END}"
 yarn install
 
 if [ "$NETWORK" == "development" ]; then
-    printf "${LOG_START}Unlocking ethereum accounts...${LOG_END}"
-    KEEP_ETHEREUM_PASSWORD=$KEEP_ETHEREUM_PASSWORD \
-        npx hardhat unlock-accounts --network $NETWORK
+  printf "${LOG_START}Unlocking ethereum accounts...${LOG_END}"
+  KEEP_ETHEREUM_PASSWORD=$KEEP_ETHEREUM_PASSWORD \
+    npx hardhat unlock-accounts --network $NETWORK
 fi
 
-printf "${LOG_START}Building contracts...${LOG_END}"
-yarn clean
-yarn build
+if [ "$SKIP_DEPLOYMENT" != true ]; then
+
+  # create tmp/ dir for fresh installations
+  rm -rf $TMP && mkdir $TMP
+
+  if [ "$THRESHOLD_NETWORK_PATH" = "" ]; then
+    cd $TMP
+    printf "${LOG_START}Cloning threshold-network/solidity-contracts...${LOG_END}"
+    # clone threshold-network/solidity-contracts as a dependency for beacon, ecdsa
+    # and tbtc
+    git clone https://github.com/threshold-network/solidity-contracts.git
+
+    cd solidity-contracts
+  else
+    printf "${LOG_START}Installing threshold-network/solidity-contracts from the local directory...${LOG_END}"
+    cd "$THRESHOLD_NETWORK_PATH"
+  fi
+
+  printf "${LOG_START}Building threshold-network/solidity-contracts...${LOG_END}"
+  yarn install && yarn clean && yarn build
+
+  # deploy threshold-network/solidity-contracts
+  printf "${LOG_START}Deploying threshold-network/solidity-contracts contracts...${LOG_END}"
+  yarn deploy --reset --network $NETWORK
+
+  yarn link
```

Comment:
`yarn link` may report a problem:
> warning There's already a package called "@threshold-network/solidity-contracts" registered. This command has had no effect. If this command was run in another folder with the same name, the other folder is still linked. Please run yarn unlink in the other folder if you want to register this folder.

We need to replace existing link, in case it's created in another directory. As yarn doesn't support a force linking (see https://github.com/yarnpkg/yarn/issues/7216) we need to workaround it.

```suggestion
  # Link the package. Replace existing link (see: https://github.com/yarnpkg/yarn/issues/7216) 
  yarn unlink || true && yarn link
```

Reply:
[daf9e3e](https://github.com/keep-network/keep-core/pull/3096/commits/daf9e3eb620e017157df9ef9db0bef35da7c4fb5)